### PR TITLE
ONS-UK-Trade-in-goods-CPA-08 

### PR DESCRIPTION
**Pull request by osamede20**
Rework done on code
Relationship between cdid dimension and observations re-defined
Cubes class added

### CODE REVIEW

**Comment by osamede20** [comment]
The latter commit d920404 is the valid one to be merged to Master.

**Review by mikeAdamss — COMMENTED** [review]
A few comment but the code is largely fine. We do need add the TransformTrace though.

File: datasets/ONS-UK-Trade-in-goods-CPA-08/main.py [hunk 1]
```diff
@@ -22,103 +38,158 @@ def right(s, amount):
...
+# In[10]:
+
 
 pd.set_option('display.float_format', lambda x: '%.0f' % x)
 
+
+# In[11]:
+
+
 df = pd.concat(tidied_sheets, ignore_index = True, sort = False).fillna('')
```
[the remark this answers is not in the document]
Comment (osamede20) [reply datasets/ONS-UK-Trade-in-goods-CPA-08/main.py:125]: Thanks Mike for the PR review and requested changes. Changes are well acknowledged and noted, and will be effected shortly.

I have looked at the TransformTrace examples on the drives, but still trying to thoroughly understand the Trace annotation. Please could you point me to one of your transforms with TransformTrace - datasets and branch name. Thanks again.

[the remark this answers is not in the document]
Comment (mikeAdamss) [reply datasets/ONS-UK-Trade-in-goods-CPA-08/main.py:125]: I need to finish now, but I'll find you some concrete examples of where we're using it and pass them over in the morning.

**Comment by osamede20** [comment]
On the latest commit 2aeed1b, the transform scraper is not picking up the correct dimension attributes - 'Product Category' and 'Product' for most of the dimension 'Product Department'. 

This can be seen from record 1417 in the CSV output data.

**Comment by canwaf** [comment]
I believe more work is required.

**Comment by osamede20** [comment]
More work is required on this trade pipeline.
Pull request was automatically merged and closed due to the issues of force-pushing commits to the master on 14 January.

A new PR for this dataset would be opened when the re-work of the stage 1 transform scraper is completed.

**Pull request merged**